### PR TITLE
Adding example custom action server

### PR DESCRIPTION
**Pull request by minhtuev**
Adding an example of Custom Action Server as a reference for developers

### CODE REVIEW

File: examples/custom_action/example_server/swagger_server/controllers/default_controller.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import connexion
+
+from ..models.data import Data  # noqa: E501
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/swagger_server/controllers/default_controller.py:3]: Why do we need lint exemptions in examples? Can we just exempt this whole directory?

File: examples/custom_action/example_server/swagger_server/models/request.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
+# coding: utf-8
+
+from __future__ import absolute_import
+from datetime import date, datetime  # noqa: F401
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/swagger_server/models/request.py:4]: The F401s are still remaining!

File: examples/custom_action/example_server/swagger_server/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# coding: utf-8
+
+# flake8: noqa
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/swagger_server/models/__init__.py:3]: Is this needed?

File: examples/custom_action/example_server/swagger_server/models/request.py (hunk 2 above)
Comment (ritvikshrivastava) [reply examples/custom_action/example_server/swagger_server/models/request.py:4]: I think these are unused imports, we can choose to remove them.

File: examples/custom_action/example_server/swagger_server/models/__init__.py (hunk 3 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/models/__init__.py]: done

File: examples/custom_action/example_server/swagger_server/models/request.py (hunk 2 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/models/request.py]: done

File: examples/custom_action/example_server/README.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Usage
+To run the server, please execute the following from the root directory:
+
+```
+pip3 install -r requirements.txt
+python3 -m swagger_server
+```
+
+Now you can make requests to hlocalhost:8080/v2/action.
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/README.md:21]:
```suggestion
Now you can make requests to localhost:8080/v2/action.
```

File: examples/custom_action/example_server/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+connexion >= 2.7.0
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/requirements.txt:1]: I saw 2.0.0 in setup.py

File: examples/custom_action/example_server/swagger_server/encoder.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+class JSONEncoder(FlaskJSONEncoder):
+    include_nulls = False
+
+    def default(self, o):
+        if isinstance(o, Model):
+            dikt = {}
+            for attr, _ in six.iteritems(o.swagger_types):
+                value = getattr(o, attr)
+                if value is None and not self.include_nulls:
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/swagger_server/encoder.py:15]:
```suggestion
                if not (value or self.include_nulls):
```

File: examples/custom_action/example_server/swagger_server/models/base_model_.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        """For `print` and `pprint`"""
+        return self.to_str()
+
+    def __eq__(self, other):
+        """Returns true if both objects are equal"""
+        return self.__dict__ == other.__dict__
+
+    def __ne__(self, other):
+        """Returns true if both objects are not equal"""
+        return not self == other
```
Comment (ritvikshrivastava) [inline examples/custom_action/example_server/swagger_server/models/base_model_.py:71]:
```suggestion
        return self != other
```

File: examples/custom_action/example_server/swagger_server/controllers/default_controller.py (hunk 1 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/controllers/default_controller.py]: This code is autogenerated by Swagger

File: examples/custom_action/example_server/swagger_server/swagger/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+  title: "MindMeld Action Server"
+basePath: "/v2"
+schemes:
+- "https"
+- "http"
+paths:
+  /action:
+    post:
+      summary: "Invoke an action "
+      description: ""
```
Comment (serapio) [inline examples/custom_action/example_server/swagger_server/swagger/swagger.yaml:15]: something should go here?

File: setup.py [hunk 9]
```diff
@@ -45,6 +45,7 @@
     "pytest-cov==2.4.0",
     "pytest-asyncio==0.8.0",
     "black>=19.10b0 ; python_version >= '3.6'",
+    "connexion>=2.0.0",
```
Comment (vijay120) [inline setup.py:48]: needs to be updated to 2.7

File: setup.py [hunk 10]
```diff
@@ -89,5 +90,6 @@
             'sentence-transformers~=0.2.6; python_version>="3.6"',
             "elasticsearch>=7.0",
         ],
+        "examples": ["connexion>=2.0.0",],
```
Comment (vijay120) [inline setup.py:93]: same, 2.7

File: examples/custom_action/example_server/requirements.txt (hunk 5 above)
Comment (vijay120) [reply examples/custom_action/example_server/requirements.txt:1]: No, the other 2.0 should be updated to 2.7s.

File: examples/custom_action/example_server/swagger_server/encoder.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+from .models.base_model_ import Model
+
+
+class JSONEncoder(FlaskJSONEncoder):
+    include_nulls = False
+
+    def default(self, o):
+        if isinstance(o, Model):
+            dikt = {}
```
Comment (vijay120) [inline examples/custom_action/example_server/swagger_server/encoder.py:12]: Please rename this var...

**Comment by minhtuev** [comment]
I am running into some problems with generating the new sample server after updating to OpenAPI 3.0, still working on this

File: examples/custom_action/example_server/swagger_server/encoder.py (hunk 11 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/encoder.py:12]: It is autogenerated by swagger

**Comment by minhtuev** [comment]
I have updated the server and the documentation to OpenAPI 3.0, you can check and test it :)

File: examples/custom_action/example_server/requirements.txt (hunk 5 above)
Comment (minhtuev) [reply examples/custom_action/example_server/requirements.txt:1]: Done

File: examples/custom_action/example_server/swagger_server/encoder.py (hunk 6 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/encoder.py:15]: Let's not modify the auto-generated code

File: examples/custom_action/example_server/swagger_server/swagger/swagger.yaml (hunk 8 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/swagger/swagger.yaml]: done

File: setup.py (hunk 9 above)
Comment (minhtuev) [reply setup.py]: done

File: setup.py (hunk 10 above)
Comment (minhtuev) [reply setup.py]: done

File: examples/custom_action/example_server/swagger_server/models/data.py [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+        self._request = request
+        self._responder = responder
+        self._action = action
+
+    @classmethod
+    def from_dict(cls, dikt) -> "Data":
+        """Returns the dict as a model
+
+        :param dikt: A dict.
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/models/data.py:44]: Let's not modify the auto-generated code

File: examples/custom_action/example_server/swagger_server/models/base_model_.py (hunk 7 above)
Comment (minhtuev) [reply examples/custom_action/example_server/swagger_server/models/base_model_.py:71]: Let's not modify the auto-generated code

**Pull request merged**